### PR TITLE
server-1786-aws | added docker_network_cidr to avoid IP overlapping in Nomad 

### PR DESCRIPTION
**Pull request by atulsingh0**
:gear: **Issue**

 To enable terraform to use specific docker network CIDR in nomad clients to avoid the IP overlapping issue between cci cloud network and nomad docker network

:white_check_mark: **Fix**

* [SERVER-1724](https://circleci.atlassian.net/browse/SERVER-1724)
* [SERVER-1786](https://circleci.atlassian.net/browse/SERVER-1786)
* [CIRCLE-39711](https://circleci.atlassian.net/browse/CIRCLE-39711)

:question: **Tests**

- Created the Nomad Client with terraform. 
- Ran Realitycheck job and checked if the docker network for running nomad jobs is falling in the `docker_network_cidr` IP range 

- [X] Passed _reality check_

### CODE REVIEW

File: nomad-aws/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ provider "aws" {
 
 module "nomad_clients" {
   # We strongly recommend pinning the version using ref=<<release tag>> as is done here
-  source = "git::https://github.com/CircleCI-Public/server-terraform.git//nomad-aws?ref=3.4.0"
+  source = "git::https://github.com/CircleCI-Public/server-terraform.git//nomad-aws?ref=3.4.1"
```
Comment (lloydChris) [inline nomad-aws/README.md:28]: There is no 3.4.1 release

**Review by lloydChris — CHANGES_REQUESTED** [review]
Can you include how you tested in the PR Description?

File: nomad-aws/README.md (hunk 1 above)
Comment (atulsingh0) [reply nomad-aws/README.md:28]: I am planning to tag once this PR is merged.

File: nomad-aws/examples/basic/main.tf [hunk 2]
```diff
@@ -51,6 +51,9 @@ module "nomad-aws" {
     module.vpc.private_subnets[0]
   ]
 
+  # cidr block you’d like to use in docker within nomad client, should not be same as subnet cidr block
+  docker_network_cidr = "192.168.0.0/16"
```
Comment (lloydChris) [inline nomad-aws/examples/basic/main.tf:55]: My comment from the [sibling PR](https://github.com/CircleCI-Public/server-terraform/pull/146#pullrequestreview-957875537) stands here as well.  Lets not include this in the basic example

File: nomad-aws/examples/irsa/main.tf [hunk 3]
```diff
@@ -55,6 +55,9 @@ module "nomad-aws" {
     module.vpc.private_subnets[0]
   ]
 
+  # cidr block you’d like to use in docker within nomad client, should not be same as subnet cidr block
+  docker_network_cidr = "192.168.0.0/16"
```
Comment (lloydChris) [inline nomad-aws/examples/irsa/main.tf:59]: Same here, This example is for highlighting IRSA not the docker settings

File: nomad-aws/main.tf [hunk 4]
```diff
@@ -5,7 +5,7 @@ resource "random_string" "key_suffix" {
 
 resource "aws_key_pair" "ssh_key" {
   count      = var.ssh_key != null ? 1 : 0
-  key_name   = "circleci-server-nomad-ssh-key-${random_string.key_suffix.result}"
+  key_name   = "${var.basename}-circleci-server-nomad-ssh-key-${random_string.key_suffix.result}"
```
Comment (lloydChris) [inline nomad-aws/main.tf:8]: Does this delete and re-create the keys?   It also seems unrelated to the new docker network parameter?

**Review by lloydChris — CHANGES_REQUESTED** [review]

**Review by lloydChris — APPROVED** [review]

**Review by ryanwohara — APPROVED** [review]
